pkgs/development/libraries/gettext/default.nix

```diff
@@ -11,7 +11,15 @@ stdenv.mkDerivation rec {
   patches = [
     ./absolute-paths.diff
     ./gettext.git-2336451ed68d91ff4b5ae1acbc1eca30e47a86a9.patch
-  ];
+  ]
+  # HACK: Since this is merely a UX fix, to minimise rebuilds only patch
+  #       the final library.
+  #       Remove the entire thing when updating to the next release.
+  ++ lib.optional (stdenv.isDarwin && !(lib.hasPrefix "bootstrap" stdenv.name))
```

Comment:
This doesn't really reduce rebuilds in a meaningful way and is kind of hacky IMHO. Introducing something like #43439 would enable moving more of this logic out of the stdenv stages, but as it stands this belongs more in the stdenv stages.